R/impute_knn.R

```diff
@@ -247,41 +247,43 @@ nn_pred <- function(index, dat) {
 
 #' @export
 bake.step_impute_knn <- function(object, new_data, ...) {
-  col_names <- purrr::map(object$columns, function(x) unname(x$x)) %>%
-    purrr::flatten_chr() %>%
-    unique()
-  check_new_data(col_names, object, new_data)
+  col_names <- purrr::map_chr(object$columns, "y")
+  all_cols <- unique(unlist(object$columns, recursive = TRUE))
```

Comment:
Why both `col_names` and `all_cols`? I had initially assumed this was to accommodate non-standard roles but that seems unrelated.

Reply:
`col_names` is the variables we are imputing, `all_cols` are the variables we are imputing + the variables we use to impute with

Reply:
And we need to make sure that all needed variables are passed to `check_new_data()`, but we just need to loop over the variables we are imputing